examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.cpp

```diff
@@ -542,14 +542,11 @@ sl_status_t show_scan_results(sl_wifi_scan_result_t * scan_result)
     for (int idx = 0; idx < (int) scan_result->scan_count; idx++)
     {
         memset(&cur_scan_result, 0, sizeof(cur_scan_result));
-
-        cur_scan_result.ssid_length = strnlen((char *) scan_result->scan_info[idx].ssid,
-                                              chip::min<size_t>(sizeof(scan_result->scan_info[idx].ssid), WFX_MAX_SSID_LENGTH));
+        cur_scan_result.ssid_length = chip::min<size_t>(strnlen((char *) scan_result->scan_info[idx].ssid, chip::DeviceLayer::Internal::kMaxWiFiSSIDLength) + 1, chip::DeviceLayer::Internal::kMaxWiFiSSIDLength); // +1 for null termination
```

Comment:
Why are we adding a +1 to the length to accommodate a null termination. That is not how Lenght of a string work.
The fact that you have a length is exactly so you don't have to use strlen and risk doing out-of-memory reads.

There seems to be a some flaws here since it is repeated in every change

Reply:
Updated the PR